### PR TITLE
Fix #27757: Make Deluge only available under Python 2

### LINKED ISSUE #27757: python3.6-deluge-1.3.13 does not install
## Issue description
python3.6-deluge-1.3.13 does not install

### Steps to reproduce
```
grantwu@allegheny:~ ⇒ nix-env -i python3.6-deluge-1.3.13

installing ‘python3.6-deluge-1.3.13’
error: pygtk-2.24.0 not supported for interpreter python3.6m
(use ‘--show-trace’ to show detailed location information)
```

## Technical details

* System: Ubuntu 16.04.2 LTS
* Nix version: nix-env (Nix) 1.11.11
* Nixpkgs version: (run `nix-instantiate --eval '<nixpkgs>' -A lib.nixpkgsVersion`): "17.09pre111388.00512470ec"
* Sandboxing enabled: (run `grep build-use-sandbox /etc/nix/nix.conf`): "grep: /etc/nix/nix.conf: No such file or directory" (What am I supposed to do here?)

### PR DESCRIPTION
**Pull request by grantwwu**
###### Motivation for this change
Fix #27757 

###### Things done
Moved Deluge out of pythonPackages into its own .nix, because it does not support Python 3 yet: http://dev.deluge-torrent.org/wiki/Plan/Python3

- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] Linux
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/top-level/all-packages.nix
Comment (FRidh) [inline pkgs/top-level/all-packages.nix]: no need for this, or does it only function with Python 2?

**Review by FRidh — APPROVED** [review]

File: pkgs/applications/networking/p2p/deluge/default.nix
Comment (FRidh) [inline pkgs/applications/networking/p2p/deluge/default.nix]: this comment is pointless if you don't say what exactly is needed

File: pkgs/top-level/all-packages.nix
Comment (grantwwu) [inline pkgs/top-level/all-packages.nix]: It only functions with Python 2; that's the point of this PR

Comment (FRidh) [inline pkgs/top-level/all-packages.nix]: Alright, then have python2Packages as parameter instead.

Comment (grantwwu) [inline pkgs/top-level/all-packages.nix]: Should all packages that are Python 2 or 3 only change to that convention?  There are a few other packages that do that, like beets, buildbot, and catfish.

**Comment by grantwwu** [comment]
@rvl I saw that you were the maintainer of Sonata, which also seems to be a Gtk application.

Have you ever encountered errors looking like this?
`deluge:31543): Gtk-WARNING **: Unable to locate theme engine in module_path: "adwaita",`

I tried to emulate Sonata's .nix but it still happens with it.

**Comment by rvl** [comment]
No, I don't remember seeing such an error.

I think you need to add `wrapGAppsHook` to `buildInputs`.

Check the `deluge` wrapper script. It should be adding themes to the `XDG_DATA_DIRS` variable.

File: pkgs/top-level/all-packages.nix
Comment (Mic92) [inline pkgs/top-level/all-packages.nix]: I think moving out of pythonPackages is reasonable because `deluge` is not used as a library, but an application from my understanding.

**Comment by grantwwu** [comment]
@rvl I see this line in the deluge wrapper script:

`export XDG_DATA_DIRS=/nix/store/2m8kkbxnch1k96vynzbpp2xnyvn3rz8c-gnome-themes-standard-3.22.1/share:/nix/store/3ypbiijmzl0v18r2bwzd7hqas8ysvxz9-adwaita-icon-theme-3.22.0/share:/nix/store/0psiqbs7h7sbc0w4d00wfx7wl4g155ab-hicolor-icon-theme-0.15/share:/nix/store/0qgr63dvaiihy25hf2lzwssqkvhv09by-cups-2.2.2/share${XDG_DATA_DIRS:+:}$XDG_DATA_DIRS
`

File: pkgs/top-level/all-packages.nix
Comment (FRidh) [inline pkgs/top-level/all-packages.nix]: The explicit version is used when it is also needed. If the package may work with both versions then  no version is used. Its kind of analogous to PEP 394 https://www.python.org/dev/peps/pep-0394/.

**Comment by rvl** [comment]
With those changes, do you still get the `Gtk-WARNING`? Does the app appear to be themed correctly? If it looks OK, maybe the Gtk warnings could be ignored.

**Comment by grantwwu** [comment]
I still got the Gtk-Warning.  I'm not 100% sure what the app is supposed to look like - I'll install Deluge from apt to check.

Also, please don't merge this yet, the Deluge devs actually told me that Deluge ought to be built against a 1.0.x series version of libtorrentRasterbar.

**Comment by grantwwu** [comment]
Result of grepping strace output:
```
12363:access("/nix/store/aqw8yr4di0hb7acmfwzjb3siay386brg-gtk+-2.24.31/lib/gtk-2.0/2.10.0/x86_64-unknown-linux-gnu/engines/libadwaita.so", F_OK) = -1 ENOENT (No such file or directory)
12364:access("/nix/store/aqw8yr4di0hb7acmfwzjb3siay386brg-gtk+-2.24.31/lib/gtk-2.0/2.10.0/x86_64-unknown-linux-gnu/engines/libadwaita.la", F_OK) = -1 ENOENT (No such file or directory)
12365:access("/nix/store/aqw8yr4di0hb7acmfwzjb3siay386brg-gtk+-2.24.31/lib/gtk-2.0/2.10.0/engines/libadwaita.so", F_OK) = -1 ENOENT (No such file or directory)
12366:access("/nix/store/aqw8yr4di0hb7acmfwzjb3siay386brg-gtk+-2.24.31/lib/gtk-2.0/2.10.0/engines/libadwaita.la", F_OK) = -1 ENOENT (No such file or directory)
12367:access("/nix/store/aqw8yr4di0hb7acmfwzjb3siay386brg-gtk+-2.24.31/lib/gtk-2.0/x86_64-unknown-linux-gnu/engines/libadwaita.so", F_OK) = -1 ENOENT (No such file or directory)
12368:access("/nix/store/aqw8yr4di0hb7acmfwzjb3siay386brg-gtk+-2.24.31/lib/gtk-2.0/x86_64-unknown-linux-gnu/engines/libadwaita.la", F_OK) = -1 ENOENT (No such file or directory)
12369:access("/nix/store/aqw8yr4di0hb7acmfwzjb3siay386brg-gtk+-2.24.31/lib/gtk-2.0/engines/libadwaita.so", F_OK) = -1 ENOENT (No such file or directory)
12370:access("/nix/store/aqw8yr4di0hb7acmfwzjb3siay386brg-gtk+-2.24.31/lib/gtk-2.0/engines/libadwaita.la", F_OK) = -1 ENOENT (No such file or directory)
12372:(deluge:24985): Gtk-WARNING **: Unable to locate theme engine in module_path: "adwaita",
```

It appears that libadwaita.so isn't being visible somehow.  Anyone know what I'm doing wrong with gnome3.gnome_themes_standard?

**Comment by rvl** [comment]
When I build it from your branch the app looks OK and I don't receive any Gtk-Warning.

**Comment by grantwwu** [comment]
I am still getting the Gtk-Warning.  Which platform are you building on?  Can you check strace?

File: pkgs/top-level/all-packages.nix
Comment (grantwwu) [inline pkgs/top-level/all-packages.nix]: I'm asking mainly what we should do with these packages:
https://github.com/NixOS/nixpkgs/blob/master/pkgs/top-level/all-packages.nix#L1223
https://github.com/NixOS/nixpkgs/blob/master/pkgs/top-level/all-packages.nix#L6667
https://github.com/NixOS/nixpkgs/blob/master/pkgs/top-level/all-packages.nix#L13524

**Comment by FRidh** [comment]
@grantwwu What is the status? Could you rebase? I'm inclined to merge it since it seems its an improvement already anyway.

**Comment by grantwwu** [comment]
Just did the squash, testing now!

**Pull request merged**
